lighthouse-core/test/audits/unused-css-rules-test.js

```diff
@@ -0,0 +1,187 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+const Audit = require('../../audits/unused-css-rules.js');
+const assert = require('assert');
+
+/* eslint-env mocha */
+
+describe('Best Practices: unused css rules audit', () => {
+  describe('#mapSheetToResult', () => {
+    let baseSheet;
+
+    function map(overrides) {
+      return Audit.mapSheetToResult(Object.assign(baseSheet, overrides));
+    }
+
+    function generate(content, length) {
+      const arr = [];
+      for (let i = 0; i < length; i++) {
+        arr.push(content);
+      }
+      return arr.join('');
+    }
+
+    beforeEach(() => {
+      baseSheet = {
+        header: {sourceURL: ''},
+        content: '',
+        used: [],
+        unused: [],
+      };
+    });
+
+    it('correctly computes percentUsed', () => {
+      assert.ok(/0%/.test(map({used: [], unused: []}).label));
+      assert.ok(/0%/.test(map({used: [], unused: [1, 2]}).label));
+      assert.ok(/50%/.test(map({used: [1, 2], unused: [1, 2]}).label));
+      assert.ok(/100%/.test(map({used: [1, 2], unused: []}).label));
+    });
+
+    it('correctly computes url', () => {
+      assert.equal(map({header: {sourceURL: ''}}).url, 'inline');
+      assert.equal(map({header: {sourceURL: 'foobar'}}).url, 'foobar');
+    });
+
+    it('correctly computes short content preview', () => {
+      const shortContent = `
+        html, body {
+          background: green;
+        }
+      `.trim();
+
+      assert.equal(map({content: shortContent}).code, shortContent);
+    });
+
+    it('correctly computes long content preview', () => {
+      const longContent = `
+        body {
+          color: white;
+        }
+
+        html {
+          content: '${generate('random', 50)}';
+        }
+      `.trim();
+
+      assert.equal(map({content: longContent}).code, `
+        body {
+          color: white;
+        } ...
+      `.trim());
+    });
+
+    it('correctly computes long rule content preview', () => {
+      const longContent = `
+        body {
+          color: white;
+          font-size: 20px;
+          content: '${generate('random', 50)}';
+        }
+      `.trim();
+
+      assert.equal(map({content: longContent}).code, `
+        body {
+          color: white;
+          font-size: 20px; ... } ...
+      `.trim());
+    });
+
+    it('correctly computes long comment content preview', () => {
+      const longContent = `
+      /**
+       * @license ${generate('a', 100)}
+       */
+      `.trim();
+
+      assert.ok(/aaa\.\.\.$/.test(map({content: longContent}).code));
+    });
+  });
+
+  describe('#audit', () => {
+    it('fails when no input present', () => {
```

Comment:
done, yeah I thought we had gotten rid of these but the test file I copied still had it in there so I thought I was misremembering haha